.github/workflows/dependency-updater.yml

```diff
@@ -0,0 +1,53 @@
+name: dependency-updater
+
+on:
+  schedule:
+    - cron: '0 12 * * *'
+  workflow_dispatch:
+      
+jobs:
+  update-latest-release:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tool: ['ytt', 'kapp', 'kbld', 'imgpkg', 'vendir']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get Current Release Version
+        id: current-version
+        run: |
+          set -eou pipefail
+          echo ::set-output name=version::$(yq eval '.${{ matrix.tool }}.version' ./hack/dependencies.yml) 
+      - name: Get Latest Release Version
+        id: latest-version
+        run: |
+          set -eou pipefail
+          echo ::set-output name=version::$(curl -sL https://api.github.com/repos/vmware-tanzu/carvel-${{ matrix.tool }}/releases/latest | jq -r '.tag_name')
+      - name: Update Dependencies File
+        if: ${{ steps.current-version.outputs.version }} != ${{ steps.latest-version.outputs.version }}
+        run: |
+          set -eou pipefail
+          
+          old_darwin_amd64_sha=$(yq eval '.${{ matrix.tool }}.checksum_darwin' ./hack/dependencies.yml) 
+          old_linux_amd64_sha=$(yq eval '.${{ matrix.tool }}.checksum_linux' ./hack/dependencies.yml)
+          
+          release_notes=$(curl https://api.github.com/repos/vmware-tanzu/carvel-${{ matrix.tool }}/releases/latest | jq .body -r)
+          new_darwin_amd64_sha=$( echo "$release_notes" | grep ${{ matrix.tool }}-darwin-amd64 | awk '{print $1;}')
+          new_linux_amd64_sha=$( echo "$release_notes" | grep ${{ matrix.tool }}-linux-amd64 | awk '{print $1;}')
+                    
+          yq eval '.${{ matrix.tool }}.version = "${{ steps.latest-version.outputs.version }}"' -i ./hack/dependencies.yml
+          yq eval ".${{ matrix.tool }}.checksum_darwin = \"${new_darwin_amd64_sha}\"" -i ./hack/dependencies.yml
+          yq eval ".${{ matrix.tool }}.checksum_linux = \"${new_linux_amd64_sha}\"" -i ./hack/dependencies.yml          
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3.12.0
```

Comment:
if we are going to be using 3rd party dependency here... (options)
- minimum: review code in their repo + lock down by digest action used
- use git cli directly in the action
- find something officially published by github (still would want to lock down by digest)